mirrord/layer/tests/fileops.rs

```diff
@@ -738,3 +714,190 @@ async fn test_go_dir_bypass(
+        .await;
+
+    layer_connection
+        .consume_xstats_then_expect_file_write("Pineapples.", 1)
+        .await;
+
+    assert!(layer_connection.is_ended().await);
+
+    // Assert all clear
+    test_process.wait_assert_success().await;
+    test_process.assert_no_error_in_stderr();
+}
+
+/// Test go file lseek.
+#[rstest]
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[timeout(Duration::from_secs(10))]
+async fn test_lseek_go(
+    #[values(Application::Go18LSeek, Application::Go19LSeek, Application::Go20LSeek)]
+    application: Application,
+    dylib_path: &PathBuf,
+) {
+    let executable = application.get_executable().await;
+    println!("Using executable: {}", &executable);
+    let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let addr = listener.local_addr().unwrap().to_string();
+    println!("Listening for messages from the layer on {addr}");
+    let mut env = get_env(dylib_path.to_str().unwrap(), &addr);
+
+    env.insert("MIRRORD_FILE_MODE", "localwithoverrides");
+    env.insert("MIRRORD_FILE_READ_WRITE_PATTERN", "/app/test.txt");
+
+    let mut test_process =
+        TestProcess::start_process(executable, application.get_args(), env).await;
+
+    let mut layer_connection = LayerConnection::get_initialized_connection(&listener).await;
+    println!("Got connection from layer.");
+
+    let fd = 1;
+    layer_connection
+        .expect_file_open_with_read_flag("/app/test.txt", fd)
+        .await;
+
+    layer_connection
+        .consume_xstats_then_expect_file_lseek(SeekFromInternal::Current(4), fd)
+        .await;
+
+    layer_connection
+        .expect_single_file_read("apples.", fd)
+        .await;
+
+    assert!(layer_connection.is_ended().await);
+
+    // Assert all clear
+    test_process.wait_assert_success().await;
+    test_process.assert_no_error_in_stderr();
+}
+
+/// Test go file access.
+#[rstest]
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+#[timeout(Duration::from_secs(10))]
+async fn test_faccessat_go(
+    #[values(
+        Application::Go18FAccessAt,
+        Application::Go19FAccessAt,
+        Application::Go20FAccessAt
+    )]
+    application: Application,
+    dylib_path: &PathBuf,
+) {
+    let executable = application.get_executable().await;
+    println!("Using executable: {}", &executable);
+    let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();
+    let addr = listener.local_addr().unwrap().to_string();
+    println!("Listening for messages from the layer on {addr}");
+    let mut env = get_env(dylib_path.to_str().unwrap(), &addr);
+
+    env.insert("MIRRORD_FILE_MODE", "localwithoverrides");
+    env.insert("MIRRORD_FILE_READ_WRITE_PATTERN", "/app/test.txt");
+
+    let mut test_process =
+        TestProcess::start_process(executable, application.get_args(), env).await;
+
+    let mut layer_connection = LayerConnection::get_initialized_connection(&listener).await;
+    println!("Got connection from layer.");
```

Comment:
Considering that this whole preamble is the same on these tests, wouldn't it be nicer to extract it to a helper function?

Reply:
extracted helper function, changed all tests where it's possible to use it.